File: app/src/main/java/jsy/sample/testwalkietalkie/utils/PathUtils.java
/*
 * Copyright (C) 2021 pedroSG94.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jsy.sample.testwalkietalkie.utils;

import android.content.ContentUris;
import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import android.os.Build;
import android.os.Environment;
import android.provider.DocumentsContract;
import android.provider.MediaStore;

import java.io.File;

/**
 * Created by pedro on 21/06/17.
 * Get absolute path from onActivityResult
 * https://stackoverflow.com/questions/33295300/how-to-get-absolute-path-in-android-for-file
 */

public class PathUtils {

  public static File getRecordPath(Context context) {
    return new File(context.getExternalFilesDir(null).getAbsolutePath()
        + "/recordFiles");
  }


}

File: app/src/main/java/jsy/sample/testwalkietalkie/utils/PathUtils.kt
package jsy.sample.testwalkietalkie.utils

import jsy.sample.testwalkietalkie.application.MyApplication
import java.io.File
import java.util.*

class PathUtils {
    companion object{
        val recordPath = File(MyApplication.instance.getExternalFilesDir(null)!!.absolutePath + "/recordFiles")

        val createFilePath = File(recordPath.absolutePath +"/" +
                Calendar.getInstance().get(Calendar.YEAR) + Calendar.getInstance().get(Calendar.MONTH) + Calendar.getInstance().get(Calendar.DATE) +
                "/" +Calendar.getInstance().get(Calendar.HOUR_OF_DAY)+"/"+ Calendar.getInstance().get(Calendar.MINUTE)/10
        )
    }
}